internal/repositories/rewards.go

```diff
@@ -0,0 +1,174 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	gmodel "github.com/DIMO-Network/identity-api/graph/model"
+	"github.com/DIMO-Network/identity-api/internal/helpers"
+	"github.com/DIMO-Network/identity-api/models"
+	"github.com/DIMO-Network/shared/dbtypes"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+	"github.com/volatiletech/sqlboiler/v4/types"
+)
+
+type RewardsCursor struct {
+	Week      int
+	VehicleID int
+}
+
+func RewardToAPI(reward models.Reward) gmodel.Earning {
+	stEarn := dbtypes.NullDecimalToInt(reward.StreakEarnings)
+	adEarn := dbtypes.NullDecimalToInt(reward.AftermarketEarnings)
+	syEarn := dbtypes.NullDecimalToInt(reward.SyntheticEarnings)
+
+	return gmodel.Earning{
+		Week:                    reward.IssuanceWeek,
+		Beneficiary:             common.BytesToAddress(reward.ReceivedByAddress.Bytes),
+		ConnectionStreak:        reward.ConnectionStreak.Ptr(),
+		StreakTokens:            stEarn,
+		AftermarketDeviceID:     reward.AftermarketTokenID.Ptr(),
+		AftermarketDeviceTokens: adEarn,
+		SyntheticDeviceID:       reward.SyntheticTokenID.Ptr(),
+		SyntheticDeviceTokens:   syEarn,
+		SentAt:                  reward.EarnedAt,
+		VehicleID:               reward.VehicleID,
+	}
+}
+
+func (r *Repository) GetEarningsByVehicleID(ctx context.Context, tokenID int) (*gmodel.VehicleEarnings, error) {
+	type rewardStats struct {
+		TokenSum   types.NullDecimal `boil:"token_sum"`
+		TotalCount int               `boil:"total_count"`
```

Comment:
My opinion is that it would be better to do this in the child resolver. It's a bit out of place here and I worry that if we introduce any filtering or anything it's gonna get lost.

Reply:
This is the bulk of what is being done in this parent resolver. Wouldn't that render the parent resolver redundant and unecessary.

Reply:
I guess I don't worry about whether a resolver has been given sufficient meaning in life. Here the resolver being invoked means we dip into the rewards table or not, that seems like enough distinction to me.

Anyway, we can leave it the way it is.